### PR TITLE
Implement Semantic Release

### PR DESCRIPTION
**Pull request by donno2048**
solves #21 
You'll have to configure `secrets.NPM_TOKEN` for it to work, and as a workflow, I can't be 100% sure it works until merged...

### CODE REVIEW

File: .github/workflows/test_release.yml [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
+    if: github.event_name == "push" && env.GITHUB_REF_NAME == "main"
+    needs: tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 'lts/*'
```
Comment (colbyfayock) [inline .github/workflows/test_release.yml:33]: how does this work with 'lts/*'? does it grab the latest lts? wondering if it makes sense to use the latest version in the matrix from tests

additionally if it's installing dependencies i think we should be using `yarn` here similar to tests to ensure we're using the same lockfile and installation in all instances

File: demo/.gitignore [hunk 2]
```diff
@@ -32,3 +32,6 @@ yarn-error.log*
 
 # vercel
 .vercel
+
+# package lock file
+package-lock.json
```
Comment (colbyfayock) [inline demo/.gitignore:37]: can you revert this? currently using yarn for the project and while i'm not opposed to change to npm with good reason, this isn't the right PR for it

Comment (donno2048) [reply demo/.gitignore:37]: It was meant for the `.gitignore` on the root but yeah, sure, I'll use yarn

File: .github/workflows/test_release.yml (hunk 1 above)
Comment (donno2048) [reply .github/workflows/test_release.yml:33]: You're right I'll use node version 16 and `yarn install`

**Comment by colbyfayock** [comment]
awesome thanks @donno2048 - we'll see how this goes after merge!

**Comment by colbyfayock** [comment]
@donno2048 i had to add some commits to get it working properly and i additionally added some configuration in `.releaserc` that i found in another community Cloudinary project, but it seems to be working now. we'll see when there's an actual code change that needs a release! thanks for the help

**Comment by colbyfayock** [comment]
@all-contributors please add @donno2048 for code

**Comment by colbyfayock** [comment]
and please email me at hello@colbyfayock.com for a form to fill out for your free swag! https://spacejelly.dev/hacktoberfest/

**Pull request merged**